compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -646,10 +646,16 @@ abstract class GenJSCode extends plugins.PluginComponent
           case js.JSSuperConstructorCall(args) =>
             implicit val pos = tree.pos
 
-            val ident =
-              origJsClass.superClass.getOrElse(sys.error("No superclass"))
-            val superTpe = jstpe.ClassType(ident.name)
-            val newTree = js.JSNew(js.LoadJSConstructor(superTpe), args)
+            val newTree = {
+              val ident =
+                origJsClass.superClass.getOrElse(sys.error("No superclass"))
+              if (args.isEmpty && ident.name == "sjs_js_Object") {
```

Comment:
No, `Definitions` is for things that are special at the IR level. `js.Object` is only special at the compiler level.